src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -241,4 +255,12 @@ public void setDeadlineTaskFilter(Predicate<? super DeadlineTask> predicate) {
         filteredDeadlineTasks.setPredicate(predicate);
     }
 
+    @Override
+    public void refilterDeadlineTaskFilteredList() {
+        Predicate<? super DeadlineTask> predicate = filteredDeadlineTasks.getPredicate();
+        assert predicate != null;
+        setDeadlineTaskFilter(null);
+        setDeadlineTaskFilter(predicate);
```

Comment:
And this shows that actually `setDeadlineTaskFilter()` is sufficient and there's no need to introduce a new thing to the interface just for it.